packages/instantsearch.js/src/middlewares/createInsightsMiddleware.ts

```diff
@@ -264,46 +245,14 @@ export function createInsightsMiddleware<
             if (existingToken && existingToken !== userToken) {
               instantSearchInstance.scheduleSearch();
             }
-
-            currentTokenType = tokenType;
-          }
-
-          // the authenticated user token cannot be overridden by a user or anonymous token
-          // for instant search query requests
-          if (
-            currentTokenType &&
-            currentTokenType === 'authenticated' &&
-            tokenType === 'default' &&
-            !unsetAuthenticatedUserToken
-          ) {
-            return;
           }
 
-          // Delay the token application to the next render cycle
-          if (!immediate) {
-            setTimeout(applyToken, 0);
-          } else {
-            applyToken();
-          }
```

Comment:
Wouldn't this be a regression to #5681?

I'm curious why only the authenticated user token required `immediate: false`.

Reply:
Good call, when I suggested removing `immediate` it was under the assumption we always called it with `true`, but indeed this branch is called with false. I wonder if instead of the `setUserTokenToSearch` having a delay, `setUserTokenToSearch` should be called with a delay inside this callback to make it clearer

Reply:
I'm not sure if the PR linked is valid atm since the double queries bug was still happening and the it was solved in another way. I'll test if removing this actually causes double queries to happen again.

But since we're removing auth token handling entirely, would we still need the immediate flag? I don't see it being used with false anywhere else.

Reply:
false is the default value, so it was called in the `setUserTokenToSearch(token)` callback, no?

Reply:
I think it was only being called with `true`: https://github.com/algolia/instantsearch/pull/6443/commits/a3a0e9c56874affff2b057e16a4b36e14fbca4cc

To be safe, we could keep this though. Might be used in the future in some way.